### ISSUE TITLE
[discussion] DirectX hooks

### ISSUE THREAD

**Issue by remcoros** [Icehunter/ffxivapp-hooker#1]
Sorry for hijacking an issue for this, but this is easier then twitter :-)

The only hook I consider 'stable' is the sharedmem variant. That is, because it serves a specific purpose: (full-screen) capturing of the rendered screen and streaming it to its client application.

I understand you're not looking for screen capture, but actual rendering. I've not really looked into that, but my advice would be to add another variant (DX9FFXIV), strip it down up until the point you only have the methods/fields/variables etc for a working hook and then slowly add things while you test.

You might want to read into the directx render pipeline a bit (and queries) to see how you can get more advanced objects rendered efficiently. (something I know little a about btw) 

Really strip out all the capture stuff if you're not going to use it, it will be less distracting and error-prone.

**Comment by Icehunter** [comment]
Yeah I'm going to add some findings here as well. I'll be adding that file in as well specifically for XIV.

Issues so far:
* Really random crashes/freezes of directx, but not consistent.
* Text is actually rendered across different areas of the screen with different transparencies/flipped. I can only assume the game actually has different rendering buffers and when using Font.Draw it just picks all.

One aspect I thought was cool was that it accidentally rendered the FPS over top a NPC name and when I rotated the camera it stayed with the NPC. It would be nice to target specific things like that (for example WoW) where I could draw something on the floor such as where to go.

I'm at least going to work through the new file, then miss-rendering after that.

Thanks for your time :)

**Comment by remcoros** [comment]
You might want to follow this thread also: https://github.com/spazzarama/Direct3DHook/issues/10
